File: Examples/Mvc.VB/Controllers/FileUploaderController.Overview.vb
Imports GleamTech.FileUltimate.AspNet
Imports GleamTech.FileUltimate.AspNet.UI

Namespace Controllers
    Partial Public Class FileUploaderController

        Public Function Overview() As ActionResult
            'Default unit is pixels. Percentage can be specified as CssLength.Percentage(100)
            Dim fileUploader = New FileUploader() With {
             .Width = 600,
             .Height = 300,
             .Resizable = True,
             .UploadLocation = "~/App_Data/Uploads"
            }

            If Request("languageSelector") IsNot Nothing Then
                fileUploader.DisplayLanguage = Request("languageSelector")
            End If

            PopulateLanguageSelector()

            Return View(fileUploader)
        End Function

        Private Sub PopulateLanguageSelector()
            ViewBag.LanguageList = New SelectList(FileUltimateWebConfiguration.AvailableDisplayCultures, "Name", "NativeName", If(Request("languageSelector"), FileUltimateWebConfiguration.CurrentLanguage.ClosestCulture.Name))
        End Sub
    End Class
End Namespace

File: Examples/AspNetCoreOnNetFullCS/Controllers/FileUploaderController.Overview.cs
using GleamTech.AspNet;
using GleamTech.FileUltimate.AspNet;
using GleamTech.FileUltimate.AspNet.UI;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.Rendering;

namespace GleamTech.FileUltimateExamples.AspNetCoreOnNetFullCS.Controllers
{
    public partial class FileUploaderController
    {
        public IActionResult Overview()
        {
            var fileUploader = new FileUploader
            {
                Width = 600, //Default unit is pixels. Percentage can be specified as CssLength.Percentage(100)
                Height = 300,
                Resizable = true,
                UploadLocation = "~/App_Data/Uploads"
            };

            var context = Hosting.GetHttpContext();
            var selectedLanguage = context.Request["languageSelector"];

            if (selectedLanguage != null)
                fileUploader.DisplayLanguage = selectedLanguage;

            PopulateLanguageSelector(selectedLanguage);

            return View(fileUploader);
        }

        private void PopulateLanguageSelector(string selectedLanguage)
        {
            ViewBag.LanguageList = new SelectList(
                FileUltimateWebConfiguration.AvailableDisplayCultures,
                "Name",
                "NativeName",
                selectedLanguage ?? FileUltimateWebConfiguration.CurrentLanguage.ClosestCulture.Name
            );
        }
    }
}
